starter-code/MyLinkedList.java

```diff
@@ -12,15 +12,45 @@ public int getSize() {
 	}
 
 	public boolean remove(int index) {
-		//to-do
+		if (index > size-1 || index < 0 || size == 0){
+			throw NullPointerException;
+			return false;
+		}else {
+			Node<T> nodeBeforeIndex = head;
+			Node<T> nodeAfterIndex;
+			for (int i = 0; i < index; i++){
+				nodeBeforeIndex = nodeBeforeIndex.getNext();
+			}
+			size--;
+			nodeAfterIndex = nodeBeforeIndex.getNext().getNext();
+			nodeBeforeIndex.setNext(nodeAfterIndex);
```

Comment:
What if `index` is zero, i.e. the caller wants to remove the head? I don't think your method will do that